lib/rust/profiling/src/frame_stats/intervals.rs

```diff
@@ -0,0 +1,147 @@
+//! This module helps to aggregate various per-frame performance statistics, collected over
+//! intervals spanning multiple GUI rendering frames.
+
+use enso_prelude::*;
+
+use crate::frame_stats;
+
+use enso_data_structures::opt_vec::OptVec;
+use enso_logger::DefaultWarningLogger as Logger;
+use enso_logger::*;
+
+
+
+// =================
+// === Intervals ===
+// =================
+
+type Intervals = OptVec<frame_stats::Accumulator>;
+
+thread_local! {
+    static ACTIVE_INTERVALS: RefCell<Intervals> = RefCell::new(Intervals::new());
+}
```

Comment:
This is a very important concept in this file and should have a detailed documentation of what it does. Docs should be provided for public things + all crucial structs / concepts.